pkg/visor/logserver/api.go

```diff
@@ -50,16 +53,52 @@ func New(log *logging.Logger, tpLogPath, localPath, customPath string, whitelist
 	if len(whitelistedPKs) > 0 {
 		authRoute.Use(whitelistAuth(whitelistedPKs))
 	}
-	// note that the survey only exists / is generated if the reward address is set
-	authRoute.StaticFile("/node-info.json", filepath.Join(localPath, "node-info.json"))
+	// note that the survey FILE only exists / is generated if the reward address is set
+	authRoute.StaticFile("/"+visorconfig.NodeInfo, filepath.Join(localPath, visorconfig.NodeInfo))
+	// This survey endpoint generates the survey as a response
+	authRoute.GET("/"+visorconfig.SurveyName, func(c *gin.Context) {
+		var rewardAddress string
+		var cAddr coincipher.Address
+		//check for reward address
+		rewardAddressBytes, err := os.ReadFile(localPath + "/" + visorconfig.RewardFile) //nolint
+		if err == nil {
+			//remove any newline from rewardAddress string
+			rewardAddress = strings.TrimSuffix(string(rewardAddressBytes), "\n")
+			//validate the skycoin address
+			cAddr, err = coincipher.DecodeBase58Address(rewardAddress)
+			if err != nil {
+				log.WithError(err).Error("Invalid skycoin reward address " + rewardAddress)
+			}
+			log.Debug("Skycoin reward address: ", cAddr.String())
+		}
+		survey, err := visorconfig.SystemSurvey()
+		if err != nil {
+			log.WithError(err).Error("Could not read system info.")
+		}
+		survey.PubKey = visorPK
+		survey.SkycoinAddress = cAddr.String()
+		// Print results.
+		s, err := json.MarshalIndent(survey, "", "\t")
+		if err != nil {
+			log.WithError(err).Error("Could not marshal system survey to json.")
+		}
+
+		c.Header("Content-Type", "application/json")
+		c.Writer.WriteHeader(http.StatusOK)
+
+		_, err = c.Writer.Write([]byte(s))
+		if err != nil {
+			httputil.GetLogger(c.Request).WithError(err).Errorf("failed to write json response")
+		}
```

Comment:
I think I mentioned this before, but you can literally just do: 

```
c.JSON(http.StatusOK, gin.H{survey})
```

Reply:
I am intentionally using generic c.Writer.Write methods instead of gin's type-specific methods.

It makes it more interchangeable with standard library http server in case we ever decide to move away from gin.